File: tech-studio-projects/financial-industry/algo-trading-platform/java/algo-trading-platform/src/com/fintech/algotrading/logging/LogInformation.java
package com.fintech.algotrading.logging;

import java.time.Instant;

public class LogInformation {

    private final LogLevel logLevel;
    private final String module;
    private final Instant now;
    private final int threadId;
    private final String threadName;

    // Constructor to initialize the log information.
    public LogInformation(LogLevel logLevel, String module, Instant now, int threadId, String threadName) {
        this.logLevel = logLevel;
        this.module = module;
        this.now = now;
        this.threadId = threadId;
        this.threadName = threadName;
    }

    // Getter for the log level.
    public LogLevel getLogLevel() {
        return logLevel;
    }

    // Getter for the module name.
    public String getModule() {
        return module;
    }

    // Getter for the time point.
    public Instant getNow() {
        return now;
    }

    // Getter for the thread ID.
    public int getThreadId() {
        return threadId;
    }

    // Getter for the thread name.
    public String getThreadName() {
        return threadName;
    }
}

File: tech-studio-projects/financial-industry/algo-trading-platform/cpp/logging/LogInformation.cpp
#include "LogInformation.h"

// Constructor to initialize the log information.
LogInformation::LogInformation(LogLevel logLevel, const std::string& module,
    const std::chrono::system_clock::time_point& now,
    int threadId, const std::string& threadName)
    : _logLevel(logLevel), _module(module), _now(now), _threadId(threadId), _threadName(threadName) {}

// Getter for the log level.
LogLevel LogInformation::GetLogLevel() const {
    return _logLevel;
}

// Getter for the module name.
std::string LogInformation::GetModule() const {
    return _module;
}

// Getter for the time point.
std::chrono::system_clock::time_point LogInformation::GetNow() const {
    return _now;
}

// Getter for the thread ID.
int LogInformation::GetThreadId() const {
    return _threadId;
}

// Getter for the thread name.
std::string LogInformation::GetThreadName() const {
    return _threadName;
}
